synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -125,28 +109,43 @@ def __init__(self, hs: "HomeServer"):
         # Whether to support MSC3772 is supported.
         self._relations_match_enabled = self.hs.config.experimental.msc3772_enabled
 
+    async def get_rules(
+        self, room_id: str, event: EventBase
+    ) -> Dict[str, List[Dict[str, dict]]]:
+        """Given an event return the rules for all users who are
+        currently in the room.
+        """
+
+        local_users = await self.store.get_local_users_in_room(room_id)
+
+        if event.type == EventTypes.Member and event.membership == Membership.JOIN:
+            if self.hs.is_mine_id(event.state_key):
+                local_users = list(local_users)
+                local_users.append(event.state_key)
+
+        ret_rules_by_user = await self.store.bulk_get_push_rules(local_users)
+
+        logger.debug("Users in room: %s", local_users)
+
+        if logger.isEnabledFor(logging.DEBUG):
+            logger.debug(
+                "Returning push rules for %r %r",
+                room_id,
+                ret_rules_by_user.keys(),
+            )
+        return ret_rules_by_user
```

Comment:
Either put one behind the log level guard or both, IMO?

.keys() will print `"dict_keys(['a'])"` which might not be what you tend. Perhaps `tuple(ret_rules_by_user)` on 134?